src/socket.c

```diff
@@ -33,6 +33,7 @@ const unsigned char compress_will   [] = { IAC, WILL, TELOPT_COMPRESS,  '\0' };
 const unsigned char compress_will2  [] = { IAC, WILL, TELOPT_COMPRESS2, '\0' };
 const unsigned char do_echo         [] = { IAC, WONT, TELOPT_ECHO,      '\0' };
 const unsigned char dont_echo       [] = { IAC, WILL, TELOPT_ECHO,      '\0' };
+const unsigned char gmcp_will       [] = { IAC, WILL, TELOPT_GMCP, SE,  '\0' };
```

Comment:
It looks like the `SE`  is unnecessary within the negotiation and causes the client (in my test case Mudlet) to try and render the line:

<img width="74" alt="screenshot 2018-03-30 12 44 35" src="https://user-images.githubusercontent.com/920019/38150146-dad234cc-341a-11e8-855d-b9429595ac27.png">

Would you update the negotiation to `{ IAC, WILL, TELOPT_GMCP, '\0' };`?

Reply:
Definitely, weird that I added it in the first place. Must have made an editing mistake or something when I was cleaning up.